libraries/tlv-account-resolution/src/account.rs

```diff
@@ -15,12 +15,17 @@ use {
 
 /// Resolve a program-derived address (PDA) from the instruction data
 /// and the accounts that have already been resolved
-fn resolve_pda(
+fn resolve_pda<'a, A, G>(
     seeds: &[Seed],
-    accounts: &[AccountMeta],
+    accounts: &[A],
     instruction_data: &[u8],
     program_id: &Pubkey,
-) -> Result<Pubkey, ProgramError> {
+    get_account_data_fn: G,
```

Comment:
Personally I hate this trait and I just tried out your suggestion and it doesn't look bad at all. Goin' in!

Since it's not a public method it's ok to be slightly complex I think